File: Bindings~/python/altunityrunner/commands/Notifications/base_notification_callbacks.py
from loguru import logger


class BaseNotificationCallbacks():
    def scene_loaded_callback(self, load_scene_notification_result):
        logger.debug("Scene {0} was loaded {1}".format(str(load_scene_notification_result.scene_name),
                                                       str(load_scene_notification_result.loadSceneMode)))

    def scene_unloaded_callback(self, scene_name):
        logger.debug("Scene {0} was unloaded".format(scene_name))

    def log_callback(self, log_notification_result):
        logger.debug("Log of type {0} with message {1} was received".format(str(log_notification_result.type),
                                                                            str(log_notification_result.message)))

    def application_paused_callback(self, application_paused):
        logger.debug("Application paused: {0}".format(application_paused))

File: Assets/AltTester/AltDriver/Notifications/BaseNotificationCallBacks.cs
using System;
using Altom.AltDriver.Logging;

namespace Altom.AltDriver.Notifications
{
    public class BaseNotificationCallBacks : INotificationCallbacks
    {
        private static readonly NLog.Logger logger = DriverLogManager.Instance.GetCurrentClassLogger();
        public void SceneLoadedCallback(AltLoadSceneNotificationResultParams altUnityLoadSceneNotificationResultParams)
        {
            logger.Log(NLog.LogLevel.Info, String.Format("Scene {0} was loaded {1}", altUnityLoadSceneNotificationResultParams.sceneName, altUnityLoadSceneNotificationResultParams.loadSceneMode.ToString()));
        }
        public void SceneUnloadedCallback(string sceneName)
        {
            logger.Log(NLog.LogLevel.Info, String.Format("Scene {0} was unloaded", sceneName));
        }
        public void LogCallback(AltLogNotificationResultParams altUnityLogNotificationResultParams)
        {
            logger.Log(NLog.LogLevel.Info, String.Format("Log of type {0} with message {1} was received", altUnityLogNotificationResultParams.level, altUnityLogNotificationResultParams.message));
        }
        public void ApplicationPausedCallback(bool applicationPaused)
        {
            logger.Log(NLog.LogLevel.Info, String.Format("Application paused: {0}", applicationPaused));
        }
    }
}